plugin/pkg/scheduler/factory/factory.go

```diff
@@ -92,22 +94,23 @@ type ConfigFactory struct {
 	// SchedulerName of a scheduler is used to select which pods will be
 	// processed by this scheduler, based on pods's annotation key:
 	// 'scheduler.alpha.kubernetes.io/name'
-	SchedulerName string
+	schedulerName string
 
 	// RequiredDuringScheduling affinity is not symmetric, but there is an implicit PreferredDuringScheduling affinity rule
 	// corresponding to every RequiredDuringScheduling affinity rule.
 	// HardPodAffinitySymmetricWeight represents the weight of implicit PreferredDuringScheduling affinity rule, in the range 0-100.
-	HardPodAffinitySymmetricWeight int
+	hardPodAffinitySymmetricWeight int
 
 	// Indicate the "all topologies" set for empty topologyKey when it's used for PreferredDuringScheduling pod anti-affinity.
-	FailureDomains string
+	failureDomains []string
 
 	// Equivalence class cache
-	EquivalencePodCache *scheduler.EquivalenceCache
+	equivalencePodCache *scheduler.EquivalenceCache
 }
 
-// Initializes the factory.
-func NewConfigFactory(client clientset.Interface, schedulerName string, hardPodAffinitySymmetricWeight int, failureDomains string) *ConfigFactory {
+// NewConfigFactory initializes the default implementation of a SchedulerConfigurationUtility.  To encourage eventual privatization of the struct type, we only
+// return the interface.
+func NewConfigFactory(client clientset.Interface, schedulerName string, hardPodAffinitySymmetricWeight int, failureDomainsCString string) scheduler.SchedulerConfigurationUtility {
```

Comment:
I don't like the name "SchedulerConfigurationUtility". First, since this is in scheduler package, we should drop Scheduler from it. I'm also not a fan of "utility" in the name. Maybe something like "Configurator" or "ConfigurationManager"?